packages/expo-modules-autolinking/src/utils.ts

```diff
@@ -15,3 +18,82 @@ export function memoize<const Fn extends (input: string, ...args: any[]) => Prom
     }
   };
 }
+
+/** List filtered top-level files in `targetPath` (returns absolute paths) */
+export async function listFilesSorted(
+  targetPath: string,
+  filter: (basename: string) => boolean
+): Promise<string[]> {
+  try {
+    // `readdir` isn't guaranteed to be sorted on Windows
+    return (await fs.promises.readdir(targetPath, { withFileTypes: true }))
+      .filter((entry) => entry.isFile() && filter(entry.name))
+      .sort((a, b) => a.name.localeCompare(b.name))
+      .map((entry) => path.join(targetPath, entry.name));
+  } catch {
+    return [];
+  }
+}
+
+/** List nested files in top-level directories in `targetPath` (returns relative paths) */
+export async function listFilesInDirectories(
+  targetPath: string,
+  filter: (basename: string) => boolean
+): Promise<string[]> {
+  const entries = (
+    await Promise.all(
+      (await fs.promises.readdir(targetPath, { withFileTypes: true }))
+        .filter((entry) => entry.isDirectory() && entry.name !== 'node_modules')
+        .sort((a, b) => a.name.localeCompare(b.name))
+        .map(async (directory) => {
+          const entries = await fs.promises.readdir(path.join(targetPath, directory.name), {
+            withFileTypes: true,
+          });
+          return entries
+            .filter((entry) => entry.isFile() && filter(entry.name))
+            .sort((a, b) => a.name.localeCompare(b.name))
+            .map((entry) => path.join(directory.name, entry.name));
+        })
+    )
+  ).flat(1);
+  return entries.sort((a, b) => a.localeCompare(b));
```

Comment:
not sure whether the sort above like `.sort((a, b) => a.name.localeCompare(b.name))` is required given we `.flat(1)` + `entries.sort((a, b) => a.localeCompare(b))` at last

Reply:
Oh I could've sworn I removed that when moving this over. I must've accidentally reverted that